wit/types.wit

```diff
@@ -30,101 +30,84 @@ interface types {
-  // which is returned when finishing an `outgoing-stream` and asychronously
-  // indicates the success or failure of writing the trailers.
-  type future-write-trailers-result = u32;
-  drop-future-write-trailers-result: func(f: future-write-trailers-result);
-  future-write-trailers-result-get: func(f: future-write-trailers-result) -> option<result<_, error>>;
-  listen-to-future-write-trailers-result: func(f: future-write-trailers-result) -> pollable;
-
   // The following block defines the `incoming-request` and `outgoing-request`
   // resource types that correspond to HTTP standard Requests. Soon, when
-  // resource types are added, the `u32` type aliases can be replaced by proper
-  // `resource` type definitions containing all the functions as methods.
-  // Later, Preview2 will allow both types to be merged together into a single
-  // `request` type (that uses the single `stream` type mentioned above). The
-  // `consume` and `write` methods may only be called once (and return failure
-  // thereafter). The `headers` and `trailers` passed into and out of requests
-  // are shared with the request, with all mutations visible to all uses.
-  // Components MUST avoid updating `headers` and `trailers` after passing a
-  // request that points to them to the outside world.
-  // The streams returned by `consume` and `write` are owned by the request and
-  // response objects. The streams are destroyed when the request/response is
-  // dropped, thus a client MUST drop any handle referring to a request/response stream
-  // before dropping the request/response or passing ownership of the request/response
-  // to the outside world. The caller can also call drop on the stream before the 
-  // request/response is dropped if they want to release resources earlier.
-  type incoming-request = u32;
-  type outgoing-request = u32;
-  drop-incoming-request: func(request: incoming-request);
-  drop-outgoing-request: func(request: outgoing-request);
-  incoming-request-method: func(request: incoming-request) -> method;
-  incoming-request-path-with-query: func(request: incoming-request) -> option<string>;
-  incoming-request-scheme: func(request: incoming-request) -> option<scheme>;
-  incoming-request-authority: func(request: incoming-request) -> option<string>;
-  incoming-request-headers: func(request: incoming-request) -> headers;
-  incoming-request-consume: func(request: incoming-request) -> result<incoming-stream>;
-  new-outgoing-request: func(
-    method: method,
-    path-with-query: option<string>,
-    scheme: option<scheme>,
-    authority: option<string>,
-    headers: headers
-  ) -> result<outgoing-request, error>;
-  outgoing-request-write: func(request: outgoing-request) -> result<outgoing-stream>;
+  // resource types are added, the `u32` type aliases can be replaced by
+  // proper `resource` type definitions containing all the functions as
+  // methods. Later, Preview2 will allow both types to be merged together into
```

Comment:
Could you remove the "when resource types are added" sentence, and then, in the next sentence, change "Preview2" to "Preview3" (a pre-existing typo)?